lambda_code/src/models/fhir_immunization.py

```diff
@@ -536,15 +84,24 @@ def remove_custom_root_validators(self, mode: Literal["pre", "post"]):
                 if "FHIRImmunizationPostValidators" in str(validator):
                     self.immunization.__post_root_validators__.remove(validator)
 
-    @timed
     def validate(self, json_data) -> Immunization:
         """Generate the Immunization model"""
+        if isinstance(json_data, str):
+            json_data = json.loads(json_data)
+
         self.set_reduce_validation_code(json_data)
-        self.add_custom_root_pre_validators()
+        self.initialize_pre_validators(json_data)
 
-        if self.add_post_validators and not self.reduce_validation_code:
-            self.add_custom_root_post_validators()
+        try:
+            self.add_custom_root_pre_validators()
```

Comment:
Would it be worth changing `add_custom_root_pre/post_validators` to `run_custom_pre/post_validators`? I don't think there's any need for the `root` anymore as they're no longer "root" validators being added to pydantinc, and we're not "adding" them to the validator any more, we're running them directly.